lib/streamlit/runtime/fragment.py

```diff
@@ -135,23 +135,16 @@ def wrapper(f: F) -> F:
 
     @wraps(non_optional_func)
     def wrap(*args, **kwargs):
-        import streamlit as st
         from streamlit.delta_generator import dg_stack
 
         ctx = get_script_run_ctx()
         if ctx is None:
             return
 
-        # TODO(vdonato): Determine whether we can just always wrap the contents of a
-        # fragment in a new container.
-        if len(dg_stack.get()) > 0:
-            dg_stack_snapshot = deepcopy(dg_stack.get())
-        else:
-            with st.container():
-                dg_stack_snapshot = deepcopy(dg_stack.get())
```

Comment:
With the changes made in #8241, I'm fairly certain we don't even need to give each fragment run its own container anymore since `dg_stack` will always contain at least the `main` `DeltaGenerator`.

Reply:
I think my main concern with not putting it into a container is with elements from the main script on the same level underneath the fragment. E.g.:

```python
@st.fragment
def my_fragment():
    some_number = st.number_input("Choose a number")
    for i in range(some_number):
        st.write(some_number)

my_fragment()
st.write("Hello world")
```

Wouldn't this just remove the `Hello world` element once you choose >= 1 in the number input?

Reply:
Ah, yeah that would be a bug introduced by removing the container entirely. I thought it would be best to get rid of it to just minimize the number of things we're doing within the fragment wrapper, but I didn't consider overwriting things on the same level. Let me re-add this.